rcl/include/rcl/publisher.h

```diff
@@ -179,7 +179,7 @@ rcl_publisher_init(
  * \return #RCL_RET_OK if publisher was finalized successfully, or
  * \return #RCL_RET_INVALID_ARGUMENT if any arguments are invalid, or
  * \return #RCL_RET_PUBLISHER_INVALID if the publisher is invalid, or
- * \return #RCL_RET_NODE_INVALID if the node is invalid, or
+ * \return #RCL_RET_NODE_INVALID if the node is not the node used to create the publisher, or
```

Comment:
IMO, `invalid` can include `not the node used to create the publisher` case. and there is a case that generally `node` is just invalid verified on `rcl_node_is_valid_except_context`? so probably current doc is good enough?

Reply:
I would actually recommend a new error code. This is conflating two things, first that the node is invalid and second that it is valid (perhaps) but not the same node that created the publisher.